modules/darwin/src/main/ts/ephox/darwin/mouse/MouseSelection.ts

```diff
@@ -29,9 +29,13 @@ export const MouseSelection = (bridge: WindowBridge, container: SugarElement<Nod
             // If a single noneditable cell is selected and the actual selection target within the cell
             // is also noneditable, make sure it is annotated
             const singleCell = boxes[0];
-            if (ContentEditable.getRaw(singleCell) === 'false' && !ContentEditable.isEditable(event.target)) {
+            const isNonEditableCell = ContentEditable.getRaw(singleCell) === 'false';
+            // Check if the target, which may be a child of the cell, needs to be explicitly selected instead of the whole cell
+            const isSelectableTarget = !Compare.eq(event.target, singleCell) && (ContentEditable.isEditable(event.target) || ContentEditable.getRaw(event.target) === 'false');
```

Comment:
This logic likely isn't going to work, if the target is say a span/strong element inside a `contenteditable="false"` element. Instead, we need to find the closest parent with `contenteditable` and see if it matches the cell. If it does, then we should select the cell. We probably need to expose the `findClosestContentEditable` function in `ContentEditable` for this.

- [ ] Fix this as mentioned above